packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -60,56 +85,56 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 		defaultValue: options[ 0 ]?.name,
 	} );
 
-	const children = options.map(
-		( { name, key, __experimentalHint, style, className } ) => {
-			const withHint = (
-				<Styled.WithHintItemWrapper>
-					<span>{ name }</span>
-					<Styled.WithHintItemHint
-					// TODO: Legacy classname. Add V1 styles are removed from the codebase
-					// className="components-custom-select-control__item-hint"
-					>
-						{ __experimentalHint }
-					</Styled.WithHintItemHint>
-				</Styled.WithHintItemWrapper>
-			);
+	const children = options.map( ( { name, key, hint, style, className } ) => {
+		const withHint = (
+			<Styled.WithHintItemWrapper>
+				<span>{ name }</span>
+				<Styled.WithHintItemHint
+				// TODO: Legacy classname. Add V1 styles are removed from the codebase
+				// className="components-custom-select-control__item-hint"
+				>
+					{ hint }
+				</Styled.WithHintItemHint>
+			</Styled.WithHintItemWrapper>
+		);
 
-			return (
-				<CustomSelectItem
-					key={ key }
-					value={ name }
-					children={ __experimentalHint ? withHint : name }
-					style={ style }
-					className={ clsx(
-						// TODO: Legacy classname. Add V1 styles are removed from the codebase
-						// 'components-custom-select-control__item',
-						className
-						// TODO: Legacy classname. Add V1 styles are removed from the codebase
-						// {
-						// 	'has-hint': __experimentalHint,
-						// }
-					) }
-				/>
-			);
-		}
-	);
+		return (
+			<CustomSelectItem
+				key={ key }
+				value={ name }
+				children={ hint ? withHint : name }
+				style={ style }
+				className={ clsx(
+					// TODO: Legacy classname. Add V1 styles are removed from the codebase
+					// 'components-custom-select-control__item',
+					className
+					// TODO: Legacy classname. Add V1 styles are removed from the codebase
+					// {
+					// 	'has-hint': hint,
+					// }
+				) }
+			/>
+		);
+	} );
 
 	const renderSelectedValueHint = () => {
 		const { value: currentValue } = store.getState();
 
-		const currentHint = options?.find(
+		const selectedOptionHint = options?.find(
 			( { name } ) => currentValue === name
-		);
+		)?.hint;
 
 		return (
 			<Styled.SelectedExperimentalHintWrapper>
 				{ currentValue }
-				<Styled.SelectedExperimentalHintItem
-				// TODO: Legacy classname. Add V1 styles are removed from the codebase
-				// className="components-custom-select-control__hint"
-				>
-					{ currentHint?.__experimentalHint }
-				</Styled.SelectedExperimentalHintItem>
+				{ selectedOptionHint && (
```

Comment:
Previously, we were rendering this component even where there wasn't any hint to show

Reply:
Good catch!